packages/openapi-generator/src/sdk-metadata/code-sample-util.ts

```diff
@@ -0,0 +1,167 @@
+import levenstein from 'fast-levenshtein';
+import { codeBlock, pascalCase } from '@sap-cloud-sdk/util';
+import { OpenApiApi, OpenApiOperation } from '../openapi-types';
+
+const distanceThreshold = 5;
+export function getMainApi(
+  serviceName: string,
+  apis: OpenApiApi[]
+): OpenApiApi {
+  let closestApi: OpenApiApi | undefined;
+  let minDistance = distanceThreshold;
+
+  apis.forEach(api => {
```

Comment:
This first part is a bit ugly and also does two things, finding the closest which is also below the threshold. Perhaps you can use reduce to buld a mapping object {distance->api} which contains only elements below the threshold. On this object you can use then Math.min on the keys to get the smallest. This does the same but uses methods which describe the flow better.

Or you wrap these lines in a helper method: findApiBelewLevensteinThreshold(apis,threshold)

Reply:
Moved the generic method to calculate and find the closest api/operation/entity/imports to generator-common.